internal/sbi/consumer/udm_service.go

```diff
@@ -61,47 +61,67 @@ func (s *nudmService) SendAuthResultToUDM(
 
 	client := s.getUdmUeauClient(udmUrl)
 
-	ctx, _, err := ausf_context.GetSelf().GetTokenCtx(models.ServiceName_NUDM_UEAU, models.NfType_UDM)
+	ctx, _, err := ausf_context.GetSelf().GetTokenCtx(models.ServiceName_NUDM_UEAU, models.NrfNfManagementNfType_UDM)
 	if err != nil {
 		return err
 	}
 
-	_, rsp, confirmAuthErr := client.ConfirmAuthApi.ConfirmAuth(ctx, id, authEvent)
-	defer func() {
-		if rspCloseErr := rsp.Body.Close(); rspCloseErr != nil {
-			logger.ConsumerLog.Errorf("ConfirmAuth Response cannot close: %v", rspCloseErr)
-		}
-	}()
+	request := &Nudm_UEAU.ConfirmAuthRequest{
+		Supi:      &id,        // Make sure this is correctly referenced
+		AuthEvent: &authEvent, // Make sure this is correctly referenced
+	}
+
+	_, confirmAuthErr := client.ConfirmAuthApi.ConfirmAuth(ctx, request)
+	if confirmAuthErr != nil {
+		logger.ConsumerLog.Errorf("Error in ConfirmAuth: %v", confirmAuthErr)
+	}
+
 	return confirmAuthErr
 }
 
 func (s *nudmService) GenerateAuthDataApi(
 	udmUrl string,
 	supiOrSuci string,
 	authInfoReq models.AuthenticationInfoRequest,
-) (*models.AuthenticationInfoResult, error, *models.ProblemDetails) {
+) (*models.UdmUeauAuthenticationInfoResult, error, *models.ProblemDetails) {
 	client := s.getUdmUeauClient(udmUrl)
 
-	ctx, pd, err := ausf_context.GetSelf().GetTokenCtx(models.ServiceName_NUDM_UEAU, models.NfType_UDM)
+	ctx, pd, err := ausf_context.GetSelf().GetTokenCtx(models.ServiceName_NUDM_UEAU, models.NrfNfManagementNfType_UDM)
 	if err != nil {
 		return nil, err, pd
 	}
 
-	authInfoResult, rsp, err := client.GenerateAuthDataApi.GenerateAuthData(ctx, supiOrSuci, authInfoReq)
+	udmAuthInfoReq := models.UdmUeauAuthenticationInfoRequest{
+		SupportedFeatures:     authInfoReq.SupportedFeatures,
+		ServingNetworkName:    authInfoReq.ServingNetworkName,
+		ResynchronizationInfo: authInfoReq.ResynchronizationInfo,
+		AusfInstanceId:        authInfoReq.AusfInstanceId,
+		CellCagInfo:           authInfoReq.CellCagInfo,
+		N5gcInd:               authInfoReq.N5gcInd,
+	}
+
+	request := &Nudm_UEAU.GenerateAuthDataRequest{
+		SupiOrSuci:                       &supiOrSuci,
+		UdmUeauAuthenticationInfoRequest: &udmAuthInfoReq,
+	}
+
+	rsp, err := client.GenerateAuthDataApi.GenerateAuthData(ctx, request)
 	if err != nil {
 		var problemDetails models.ProblemDetails
-		if authInfoResult.AuthenticationVector == nil {
+		if rsp.UdmUeauAuthenticationInfoResult.AuthenticationVector == nil {
```

Comment:
If `err != nil`, `rsp` may be nil, and line 111 will cause "panic: runtime error: invalid memory address or nil pointer dereference."

Reply:
That's a huge mistake. Thank you.